docs/crds/ssorule.md

```diff
@@ -0,0 +1,404 @@
+
+The `cookieName` property can be used to express the name of HTTP cookie for an authorized session. If this field is not specified, NSX ALB will create the authorized session cookie with a random name.
+
+```yaml
+  cookieName: OAUTH_XYZ
+```
+
+#### Express timeout for authorized session cookie
+
+The `cookieTimeout` property can be used to specify the HTTP cookie timeout in **minutes** for an authorized session. It supports values from **1** to **1440** and defaults to **60**.
+
+```yaml
+  cookieTimeout: 120
+```
+
+#### Express Logout URI for OAuth
+
+The `logoutURI` property can be used to express the URI that will trigger OAuth logout.  
+
+```yaml
+  logoutURI: https://my-ssorule.test.com/oauth/logout
+```
+
+#### Express Application and IDP settings for OAuth/OIDC
+
+The `oauthSettings` property can be used to express the application specific OAuth configuration and Identity Provider (IDP) settings for OAuth/OIDC.  
+The application specific properties are specified as a list of `appSettings` properties.
+
+#### Express Client ID
+
+The `clientID` property can be used to express the client ID for the application. It is an application specific identifier registered with the authorization server or Identity Provider (IDP).
+
+```yaml
+  clientID: my-client-id
+```
+
+#### Express Client Secret
+
+The `clientSecret` field can be used to express the client secret for the application. It is an application specific identifier secret registered with the authorization server, or IDP. Since `clientSecret` is a sensetive field in NSX ALB, AKO requires it to be specified inside a Kubernetes Secret object. So, this `clientSecret` field should be the name of the Kubernetes secret object that specifies the actual client secret value (Base64 encoded) in the **clientSecret** data field.
+
+```yaml
+  clientSecret: my-oauth-secret
+```
+
+A sample Kubernetes secret object, with the actual value for client and secret (Base64 encoded) in the **clientSecret** data field is shown below.  
+
+```yaml
+  apiVersion: v1
+  data:
+    clientSecret: bXktY2xpZW50LXNlY3JldA==
+    serverSecret: bXktc2VydmVyLXNlY3JldA==
+  kind: Secret
+  metadata:
+    name: my-oauth-secret
+  type: Opaque
+```
+
+#### Express OpenID Connect configuration
+
+The `oidcConfig` property can be used to express OpenID Connect specific configuration.
+
+```yaml
+  oidcConfig:
+    oidcEnable: true
+    profile: true
+    userinfo: true
+```
+`oidcEnable`, if set to **true**, adds OpenID as one of the scopes enabling the OpenID Connect flow.  
+`profile`, if set to **true**, will allow fetching profile information by enabling profile scope. It defaults to **true**.   
+`userinfo`, if set to **true**, will allow fetching profile information from the Userinfo Endpoint.
+
+#### Express Scope for OAuth
+
+The `scopes` property can be used to express the scope for OAuth to give limited access to the application.
+
+```yaml
+  scopes: ["scope1"]
```

Comment:
what is this `scope1` value present? Is it already created one at Oauth server side or any random value?

Reply:
This is the scope that the application requests from the authorization server. Scopes are already created on the authorization server and are associated with different levels of access to the resources. The application has to specify a valid scope.